lib/ae_mdw/channels.ex

```diff
@@ -113,22 +120,37 @@ defmodule AeMdw.Channels do
            amount: amount,
            updates: [{{last_updated_height, _mbi}, last_updated_txi} | _rest] = updates
          ),
-         is_active?
+         is_active?: is_active?,
+         node_details?: node_details?
        ) do
-    %{"hash" => tx_hash, "tx" => %{"type" => tx_type}} = Txs.fetch!(state, last_updated_txi)
-
-    %{
-      channel: Enc.encode(:channel, channel_pk),
-      initiator: Enc.encode(:account_pubkey, initiator_pk),
-      responder: Enc.encode(:account_pubkey, responder_pk),
-      state_hash: :aeser_api_encoder.encode(:state, state_hash),
-      amount: amount,
+    %{"block_hash" => block_hash, "hash" => tx_hash, "tx" => %{"type" => tx_type}} =
+      Txs.fetch!(state, last_updated_txi)
+
+    channel = %{
+      channel: encode(:channel, channel_pk),
+      initiator: encode_account(initiator_pk),
+      responder: encode_account(responder_pk),
+      state_hash: encode(:state, state_hash),
       last_updated_height: last_updated_height,
       last_updated_tx_hash: tx_hash,
       last_updated_tx_type: tx_type,
       updates_count: length(updates),
       active: is_active?
     }
+
+    with true <- node_details?,
```

Comment:
Is there a reason why we're unable to add this `get_channel_at_hash` information on the `/v2/channels` endpoint for each channel?

Reply:
It would be better after a profiling to check the impact on memory and performance since it needs to load multiple block state trees for different block hashes (one for each channel updated on different micro block).

Reply:
It demands profiling to check the impact on memory and performance since it needs to load multiple block state trees for different block hashes (one for each channel updated on different micro block). On a `/v2/channels` refactoring would be a better moment for that.